src/store/state/utils.ts

```diff
@@ -18,3 +18,21 @@ export const generateBudgetYear = (year: number): BudgetYear => ({
   income: [generateBudgetGroup('General', ['Other'])],
   expenses: [generateBudgetGroup('General', ['Other'])]
 });
+
+export const generateBudgetYearFromCurrent = (year: number, current: BudgetYear): BudgetYear => {
+  function clearValues(value: BudgetGroup) {
+    for (const budget of value.budgets) {
+      budget.values = new Array(12).fill(0);
+    }
+    return value;
+  }
+
+  current.income = current.income.map(clearValues);
+  current.expenses = current.expenses.map(clearValues);
```

Comment:
You're passing the current budget year via `getCurrentYear` into this and re-assigning the values of it, clearing the income and expenses of the current years with it. You probably don't want to re-assign it but pass this directly into the return statement like this:

```ts
  return {
    year,
    income: current.income.map(clearValues),
    expenses: current.expenses.map(clearValues)
  };
```